### ISSUE TITLE
dependency resolution error

### ISSUE THREAD

**Issue by melezhik** [berkshelf/berkshelf#1670]
Hi!

Now, when berks fail to resolve dependencies it just dumps out a cookbooks version list and does not tell anything about which cookbooks cause a dependency restitution error . So it's pain to troubleshoot. Any plans on improving it?

**Comment by melezhik** [comment]
Hi! Any news here ?

**Comment by timlianov** [comment]
Any updates here, guys?

**Comment by kplimack** [comment]
try adding this to your Berksfile
```
solver :ruby, :preferred
```

**Comment by thommay** [comment]
Hi, sorry for the delay in replying.
We have no real plans to work on feature requests for Berkshelf right now, especially during the Chef Client 13 cycle. We'd gladly accept pull requests that improve this, though.

**Comment by melezhik** [comment]
Yeah, it's just annoying thing about berks which makes it hard to use it in troubleshooting, don't you think so?

**Comment by tbonci** [comment]
Can you provide examples of some berksfiles showing the failures and their output? This would help in building a test case for someone to code against.

**Comment by melezhik** [comment]
hi @tbonci  will try to add some input  soon ...

**Comment by kplimack** [comment]
if you change the solver you use, the output is much more helpful.
add this to your Berksfile and try depsolving again.
```
solver :ruby, :preferred
```
not only is it faster, but it only shows you the things that are failing.  its made my life a lot easier

**Comment by melezhik** [comment]
Hi @kplimack  thanks , will try you hint ...

**Comment by melezhik** [comment]
Well, upon @cheesepluse request from chef slack community adding here a simple user story


```

vagrant@Debian-jessie-amd64-netboot:~/my/issue-1670$ ls -l
total 12
drwxr-xr-x 10 vagrant vagrant 4096 Mar 23 15:04 baz
-rw-r--r--  1 vagrant vagrant  104 Mar 23 15:13 Berksfile
drwxr-xr-x 10 vagrant vagrant 4096 Mar 23 15:04 foo
vagrant@Debian-jessie-amd64-netboot:~/my/issue-1670$ cat baz/metadata.rb

name             'baz'
maintainer       'YOUR_COMPANY_NAME'
maintainer_email 'YOUR_EMAIL'
license          'All rights reserved'
description      'Installs/Configures baz'
long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
version          '0.1.0'


depends 'yum-epel', '~> 2.1.1'
vagrant@Debian-jessie-amd64-netboot:~/my/issue-1670$ cat foo/metadata.rb
name             'foo'
maintainer       'YOUR_COMPANY_NAME'
maintainer_email 'YOUR_EMAIL'
license          'All rights reserved'
description      'Installs/Configures foo'
long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
version          '0.1.0'


depends "bar"

$ cat Berksfile
source 'https://supermarket.chef.io'

cookbook 'foo', :path => './foo'
cookbook 'baz', :path => './baz'
```

Now I run "berks" to resolve my dependencies.  BTW cookbook bar https://supermarket.chef.io/cookbooks/bar#dependencies has compat_resource = 12.14.7 which in conflict with  "depends 'yum-epel', '~> 2.1.1'" required at baz cookbook 

So let's go:

```
$ berks
Resolving cookbook dependencies...
Fetching 'baz' from source at baz
Fetching 'foo' from source at foo
Fetching cookbook index from https://supermarket.chef.io...
Unable to satisfy the following requirements:

- `compat_resource (= 12.14.7)` required by `bar-0.1.0`
- `compat_resource (>= 12.16.3)` required by `yum-epel-2.1.1`
Unable to find a solution for demands: baz (0.1.0), foo (0.1.0)

```

It would be nice to have a full dependency chain resulted in conflict, something like that ( see extra chunks at both chains ):


- `compat_resource (= 12.14.7)` required by `bar-0.1.0` required by `foo-0.1.0`
- `compat_resource (>= 12.16.3)` required by `yum-epel-2.1.1` required by `baz-0.1.0` required by `bar-0.1.0`


BWT this is quite simple examples with a few cookbooks on the top level. Having in real life much lager dependencies chains with multiple nesting it is even harder to investigate the  dependency resolution issues with current berks out format

**Comment by melezhik** [comment]
And yeah, adding `solver :ruby, :preferred` to Berksfile does not change  the picture. The same output.

**Comment by melezhik** [comment]
and yeah my env is : 

```
$ berks --version
5.0.0
```

**Comment by rchekaluk** [comment]
Seems like a duplicate of https://github.com/chef/dep-selector-libgecode/issues/43
